src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,73 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> inctances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
-    public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+    public Object getInstance(Class<?> interfaceClass) {
+        Class<?> instanceClass = findImplamentation(interfaceClass);
+        if (!instanceClass.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("This class doesn't have annotation "
+                    + Component.class.getName());
+        }
+        Object classImplamentationsIntance = null;
+        Field[] declaredClassFields = instanceClass.getDeclaredFields();
+        for (Field field : declaredClassFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldClassInstance = getInstance(field.getType());
+                classImplamentationsIntance = createNewInstance(instanceClass);
+                field.setAccessible(true);
```

Comment:
put this line inside the try block, cause `setAccessible()` would throw an exception too. [See documentation](https://docs.oracle.com/javase/7/docs/api/java/lang/reflect/AccessibleObject.html#setAccessible(boolean))